training/deepspeech_training/util/audio.py

```diff
@@ -358,12 +371,44 @@ def read_duration(audio_type, audio_file):
     raise ValueError('Unsupported audio type: {}'.format(audio_type))
 
 
-def pcm_to_np(audio_format, pcm_data):
-    _, channels, width = audio_format
-    if width not in [1, 2, 4]:
-        raise ValueError('Unsupported sample width: {}'.format(width))
-    dtype = [None, np.int8, np.int16, None, np.int32][width]
+def get_dtype(audio_format):
+    if audio_format.width not in [1, 2, 4]:
+        raise ValueError('Unsupported sample width: {}'.format(audio_format.width))
+    return [None, np.int8, np.int16, None, np.int32][audio_format.width]
+
+
+def pcm_to_np(pcm_data, audio_format=DEFAULT_FORMAT):
+    assert audio_format.channels == 1  # only mono supported for now
+    dtype = get_dtype(audio_format)
     samples = np.frombuffer(pcm_data, dtype=dtype)
-    assert channels == 1  # only mono supported for now
     samples = samples.astype(np.float32) / np.iinfo(dtype).max
     return np.expand_dims(samples, axis=1)
+
+
+def np_to_pcm(np_data, audio_format=DEFAULT_FORMAT):
+    assert audio_format.channels == 1  # only mono supported for now
+    dtype = get_dtype(audio_format)
+    np_data = np_data.squeeze()
+    np_data = np_data * np.iinfo(dtype).max
+    np_data = np_data.astype(dtype)
+    return np_data.tobytes()
+
+
+def rms_to_dbfs(rms):
+    return 20.0 * math.log10(max(1e-16, rms)) + 3.0103
+
+
+def mean_dbfs(sample_data):
+    return rms_to_dbfs(math.sqrt(np.mean(np_capped_squares(sample_data))))
+
+
+def max_dbfs(sample_data):
+    return rms_to_dbfs(max(abs(np.min(sample_data)), abs(np.max(sample_data))))
```

Comment:
For a single sample (in this case the one with the highest amplitude) the _root mean square_ is the same as its absolute value.

Reply:
I'm more confused by why you're computing max dbFS from a single peak sample. Shouldn't this be a true RMS measurement over windows of the signal? Maybe add a comment to clarify why the unusual choice?

Reply:
Even in a previous version of the PR you were computing this over windows, so clearly you've made the explicit decision to use the peak measurement. A comment or docstring explaining would help.

Reply:
Yeah - you are right regarding my early version. It turned out that the "normalized" samples (based on the old time-interval way to measure dBFS) were always running into overdrive, as the highest energy samples were capped to `(-1, 1)`. This is because the mean loudness of a certain time-slice will typically not be in the same value range as its peak amplitude samples.